x-pack/plugins/security_solution/public/common/components/guided_onboarding_tour/tour.tsx

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ReactChild } from 'react';
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
+import useObservable from 'react-use/lib/useObservable';
+import { catchError, of, timeout } from 'rxjs';
+import { useKibana } from '../../lib/kibana';
+import { securityTourConfig, SecurityStepId } from './tour_config';
+
+export interface TourContextValue {
+  activeStep: number;
+  endTourStep: (stepId: SecurityStepId) => void;
+  incrementStep: (stepId: SecurityStepId, step?: number) => void;
+  isTourShown: (stepId: SecurityStepId) => boolean;
+}
+
+const initialState: TourContextValue = {
+  activeStep: 0,
+  endTourStep: () => {},
+  incrementStep: () => {},
+  isTourShown: () => false,
+};
+
+const TourContext = createContext<TourContextValue>(initialState);
+
+export const TourContextProvider = ({ children }: { children: ReactChild }) => {
+  const { guidedOnboardingApi } = useKibana().services.guidedOnboarding;
+
+  const isRulesTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.rules).pipe(
+      // if no result after 30s the observable will error, but the error handler will just emit false
+      timeout(30000),
+      catchError((error) => of(false))
+    ) ?? of(false),
+    false
+  );
+  const isAlertsCasesTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.alertsCases).pipe(
+      // if no result after 30s the observable will error, but the error handler will just emit false
+      timeout(30000),
+      catchError((error) => of(false))
+    ) ?? of(false),
+    false
+  );
+
+  const tourStatus = useMemo(
+    () => ({
+      [SecurityStepId.rules]: isRulesTourActive,
+      [SecurityStepId.alertsCases]: isAlertsCasesTourActive,
+    }),
+    [isRulesTourActive, isAlertsCasesTourActive]
+  );
+
+  const isTourShown = useCallback((stepId: SecurityStepId) => tourStatus[stepId], [tourStatus]);
+  const [activeStep, _setActiveStep] = useState<number>(1);
+
+  const incrementStep = useCallback((stepId: SecurityStepId, step?: number) => {
+    _setActiveStep((prevState) =>
+      step != null && step <= securityTourConfig[stepId].length
```

Comment:
Shouldn't it be `step < securityTourConfig[stepId].length`?

Reply:
> Shouldn't it be step < securityTourConfig[stepId].length?

no, say there are 5 steps. say we are on step 4. the method is called like `incrementStep(SecurityStepId.alertsCases, 5)`. We need to set it to 5

> I saw the step parameter is never being passed

Other teams will use this, and in a different approach i had used the argument. i think i can remove it

Reply:
ok removed it, thanks!